ethcore/src/engines/clique/mod.rs

```diff
@@ -168,7 +168,7 @@ pub struct Clique {
 	block_state_by_hash: RwLock<LruCache<H256, CliqueBlockState>>,
 	proposals: RwLock<HashMap<Address, VoteType>>,
 	signer: RwLock<Option<Box<EngineSigner>>>,
-	step_service: Option<Arc<StepService>>,
+	step_service: Option<StepService>,
```

Comment:
If we disallow `period=0` we could get rid of the `Option` here as well.

Reply:
Yes, that's planned for the follow up PR. :)